api/views.py

```diff
@@ -0,0 +1,161 @@
+            return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def summary_cost_of_salaries(self, request, *args, **kwargs):
+        sum_of_salaries = Employee.objects.aggregate(sum_of_salaries=Sum('salary'))
+        return Response("Summary cost of all salaries: " + str(sum_of_salaries['sum_of_salaries']))
+
+
+class IncomeViewSet(viewsets.ModelViewSet):
+    queryset = Income.objects.all()
+    serializer_class = IncomeSerializer
+    filter_fields = ('name', 'date')
+
+    @action(detail=False, methods=['get'])
+    def last_incomes(self, request, *args, **kwargs):
+        number_of_incomes_to_show = request.GET.get('number_of_incomes')
+        incomes = Income.objects.order_by('date')
+        if number_of_incomes_to_show:
+            incomes = incomes[:int(number_of_incomes_to_show)]
+
+        serializer = EventSerializer(incomes, many=True)
+        return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def sum(self, request, *args, **kwargs):
+        days_back = request.GET.get('days', 7)
+        if str(days_back).isdigit():
+            current_date = datetime.today().date()
+            date_for_filter = current_date - timedelta(days=int(days_back))
+            sum_of_incomes = Income.objects.filter(date__gte=date_for_filter).aggregate(sum_of_incomes=Sum('sum'))[
+                'sum_of_incomes']
+            if not sum_of_incomes:
+                sum_of_incomes = "0"
+            return Response("Total income from the last " + str(days_back) + " days is: " + str(sum_of_incomes))
+        return Response("days parameter should be a number")
+
+
+class OutcomeViewSet(viewsets.ModelViewSet):
+    queryset = Outcome.objects.all()
+    serializer_class = OutcomeSerializer
+    filter_fields = ('name', 'date')
+
+    @action(detail=False, methods=['get'])
+    def last_outcomes(self, request, *args, **kwargs):
+        number_of_outcomes_to_show = request.GET.get('number_of_outcomes', None)
+        outcomes = Outcome.objects.order_by('date')[:int(number_of_outcomes_to_show)]
+        if number_of_outcomes_to_show:
+            outcomes = outcomes[:int(number_of_outcomes_to_show)]
+        serializer = EventSerializer(outcomes, many=True)
+        return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def sum(self, request, *args, **kwargs):
+        days_back = request.GET.get('days', 7)
+        if str(days_back).isdigit():
+            current_date = datetime.today().date()
+            date_for_filter = current_date - timedelta(days=int(days_back))
+            sum_of_outcomes = Outcome.objects.filter(date__gte=date_for_filter).aggregate(sum_of_outcomes=Sum('sum'))[
+                'sum_of_outcomes']
+            if not sum_of_outcomes:
+                sum_of_outcomes = "0"
+            return Response(
+                "Total outcome from the last " + str(days_back) + " days is: " + str(sum_of_outcomes))
```

Comment:
Minor thing: in Python 3.7 you can format strings like this: `f"Total outcome from the last {str(days_back)} days is {str(sum_of_outcomes)}"`

More info: https://realpython.com/python-f-strings/#f-strings-a-new-and-improved-way-to-format-strings-in-python

Reply:
Another thing - how about returning response in JSON format here?